content/en/blog/_posts/2024-11-04-kubernetes-1.32-sneak-peek.md

```diff
@@ -0,0 +1,133 @@
+
+
+**DRA Structured parameters** 
+
+The original dynamic resource allocation (DRA) uses claim and class parameters that are opaque to Kubernetes. KEP [#4381](https://github.com/kubernetes/enhancements/issues/4381) introduces structured parameters so that kube-scheduler and Cluster Autoscaler can handle a claim allocation themselves without relying on a third-party driver.
+As users are increasingly adopting Kubernetes as their management solution for batch-processing workloads and edge computing, the need of exposing specialized hardware to Pods is increasing.
+Such workloads no longer need just RAM and CPU - so this KEP introduced a new API for describing which of these specialized resources a Pod needs, for example devices like a GPU or other kinds of accelerators.
+
+
+**Prioritized Alternatives in Device Requests** 
+
+KEP [#4816](https://github.com/kubernetes/enhancements/issues/4816) adds support for a prioritized list of selection criteria, attached to a device request in a `ResourceClaim`.
+The DRA's Structured Parameters feature has added the ability to make requests for very specific types of devices using a `ResourceClaim`.
+
+However, the current API did not allow the user to indicate any priority when multiple types or configurations of devices might have met the needs of the workload. This feature allows the user to specify alternative requests that satisfy the workloads need, giving more flexibility to the scheduler in the workloads' scheduling phase.
+
+
+**Add Resource Health Status to the Pod Status**
+
+When a Pod is using a device that has failed or is temporarily unhealthy, it is difficult to know. KEP [#4680](https://github.com/kubernetes/enhancements/issues/4680) makes troubleshooting of Pod crashes easier by exposing device health via `Pod Status`.
+
+
+### Windows strikes back!
+
+KEP [#4802](https://github.com/kubernetes/enhancements/issues/4802) adds support for graceful shutdown of Windows nodes in Kubernetes clusters.
+Currently, Kubernetes provides graceful node shutdown functionality for Linux nodes but lacked equivalent support for Windows. This enhancement aims to enable the kubelet on Windows nodes to handle system shutdown events properly. By doing so, it ensures that Pods running on Windows nodes are gracefully terminated, allowing workloads to be rescheduled without disruption. This improvement enhances the reliability and stability of clusters that include Windows nodes, especially during a planned maintenance or any system updates.
+
+KEP [#4885](https://github.com/kubernetes/enhancements/issues/4885) adds CPU and memory affinity support to Windows nodes in Kubernetes, thus improving performance for workloads that benefit from cache vicinity and NUMA optimizations, this effort also aligns Windows capabilities with Linux nodes.
+
```

Comment:
A few more exciting things from SIG Node around quility of life and small improvements:

1. https://github.com/kubernetes/kubernetes/pull/127566 - support for systemd watchdog is implemented for the kubelet. 
2. Better message in image pull backoff message: https://github.com/kubernetes/kubernetes/pull/127918

Also we are preparing to the major GA in 1.33 - tracking last remaining items and feedback for sidecar containers KEP: https://github.com/kubernetes/enhancements/issues/753#issuecomment-2350136594

Reply:
Hey @SergeyKanzhelev 👋 

Thanks for pointing these out, amazing enhancements _huge fan of the better image pull backoff error btw_
Just so you know this blog being the sneak peek means that the KEP list chosen to be highlighted is non-exhaustive by definition, it's an arbitrary subset by the Release Team (Enhancements, Comms).

You might want to add these to the [Release Highlights discussion](https://github.com/kubernetes/sig-release/discussions/2639) for SIG Node instead, those and any KEPs highlighted by the SIG leadership will be included in the final Release Blog!

Reply:
sounds good. https://github.com/kubernetes/sig-release/discussions/2639#discussioncomment-11093925

Reply:
Amazing, thank you!